### ISSUE TITLE
how can a single coturn can handle multiple synapse servers

### ISSUE THREAD

**Issue by Puvendhan** [coturn/coturn#1233]
Hi Team,

Could you please help me on the below configuration

how can a single coturn can handle multiple synapse servers?
One coturn should accept multiple shared secrets? 
Coturn authentication with multiple shared secrets for each synapse, ?

if we have multiple synapse servers, coturn should authenticate with each synapse server with different shared secrets, how to implement the setup?

**Comment by Puvendhan** [comment]
Hi Team, Can anyone look into this.?

**Comment by Puvendhan** [comment]
Hi Team,

I am getting the below error while running the turnadmin command, can anyone take a look?

`# turnadmin -s ksdjfksjdlfklsdjfksajkfkdjalfkls5 -r sample.com --psql-userdb=postgresql://sample:sample@dbrul:5432/dbname
0: : PostgreSQL connection was closed.
0: : log file opened: /var/log/turn_16302_2023-07-13.log`

`turnadmin -s ksdjfksjdlfklsdjfksajkfkdjalfkls5 -r sample.com --psql-userdb="host=dburl dbname=sample user=sample password=sample connect_timeout=30"`

**Comment by eakraly** [comment]
Hi @Puvendhan 
If I understand your question correctly you want to use multiple credentials with on turnserver
I do not have any experience with synapse but it seems you figured it out - storing multiple credentials in a DB seems like one way to do it

**Comment by Puvendhan** [comment]
Yes, right. i followed the below steps , but i am getting error. is there any other documentation available to setup the postgres DB?

https://github.com/coturn/coturn/blob/master/docs/PostgreSQL.md


`#turnserver --psql-userdb=postgresql://sample:sample@dburl:5432/coturn (the command is successful and able to connect to postgres db)

Output:
0: : PostgreSQL DB connection success: postgresql://sample:sample@dburl:5432/coturn


**I am getting error while running the below commands which are mentioned in the above URI,**

#turnadmin -s secretkey -r coturn.sample.xyz.com --psql-userdb="postgresql://sample:sample@dburl:5432/coturn"
0: : PostgreSQL connection was closed.
0: : log file opened: /var/log/turn_7855_2023-07-14.log

# cat /var/log/turn_7855_2023-07-14.log
0: : log file opened: /var/log/turn_7855_2023-07-14.log
0: : PostgreSQL connection was closed.

#turnadmin -s secretkey -r coturn.sample.xyz.com -e "host=dburl dbname=coturn user=sample password=sample"
0: : PostgreSQL connection was closed.
0: : log file opened: /var/log/turn_7823_2023-07-14.log
`

**Comment by Puvendhan** [comment]
Hi Team,

Any update on this,? We are completely blocked, can anyone take a look?

**Comment by eakraly** [comment]
Hi @Puvendhan 
Have you been able to resolve the issue? It seems very much specific to your setup - make sure you use the right db url,username,password,dbname etc

**Comment by eakraly** [comment]
I see you opened #1251 so I guess you did figure it out?
